olive/evaluator/olive_evaluator.py

```diff
@@ -275,6 +275,172 @@ def _evaluate_latency(
 
         return OliveEvaluator.compute_latency(metric, latencies)
 
+    @staticmethod
+    def _evaluate_distributed_accuracy_worker(config) -> Tuple[List[Any], List[Any]]:
+        model_path = config["model_path"]
+        local_rank = config["local_rank"]
+        world_size = config["world_size"]
+        inference_settings = config.get("inference_settings", {}) or {}
+        metric = Metric.from_json(config["metric"])
+        dataloader, _, post_func = OnnxEvaluator.get_user_config(metric)
+
+        import os
+
+        os.environ["OMPI_COMM_WORLD_RANK"] = str(local_rank)
+        os.environ["OMPI_COMM_WORLD_SIZE"] = str(world_size)
+
+        from mpi4py import MPI
+
+        local_rank = MPI.COMM_WORLD.Get_rank()
+
+        inference_settings["execution_provider"] = ["CUDAExecutionProvider", "CPUExecutionProvider"]
+        inference_settings["provider_options"] = [{"device_id": str(local_rank)}, {}]
```

Comment:
The EP should be selected based on the accelerator_spec provided by the Engine to the evaluate_method. Right now it does not, so please add a FIXME note here.

Reply:
Do we have the accelerator_spec functionality in the current implementation or is this something of a TODO?

Reply:
Done!